src/Features/Core/Portable/MakeClassAbstract/AbstractMakeClassAbstractCodeFixProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.MakeClassAbstract
+{
+    internal abstract class AbstractMakeClassAbstractCodeFixProvider<TClassDeclarationSyntax> : SyntaxEditorBasedCodeFixProvider
+        where TClassDeclarationSyntax : SyntaxNode
+    {
+        internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.Compile;
+
+        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            if (context.Diagnostics.Length == 1 &&
+                IsValidMemberNode(context.Diagnostics[0].Location?.FindNode(context.CancellationToken)))
+            {
+                context.RegisterCodeFix(
+                    new MyCodeAction(c => FixAsync(context.Document, context.Diagnostics[0], c)),
+                    context.Diagnostics);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        protected sealed override Task FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, SyntaxEditor editor,
+            CancellationToken cancellationToken)
+        {
+            for (var i = 0; i < diagnostics.Length; i++)
+            {
+                var declaration = diagnostics[i].Location?.FindNode(cancellationToken);
+
+                if (IsValidMemberNode(declaration))
+                {
+                    var enclosingClass = declaration.FirstAncestorOrSelf<TClassDeclarationSyntax>();
```

Comment:
`IsValidMemberNode` is already looking for the enclosing class but I wasn't a big fan of returning the class and a boolean from it so I have decided to duplicate the call.